esti/s3_gateway_test.go

```diff
@@ -54,6 +55,69 @@ func newMinioClient(t *testing.T, getCredentials GetCredentials) *minio.Client {
 	return client
 }
 
+func TestS3UploadToReadOnlyRepoError(t *testing.T) {
+	ctx, _, repo := setupTest(t)
+	defer tearDownTest(repo)
+
+	readOnlyRepo := createReadOnlyRepositoryByName(ctx, t, "tests3uploadobjectdestreadonly")
+	defer deleteRepositoryIfAskedTo(ctx, readOnlyRepo)
+
+	minioClient := newMinioClient(t, credentials.NewStaticV4)
+	// create a file that's larger than any internal buffering minio does
+	const tenMBInBytes = 10 * 1024 * 1024
+	tmpFile, cleanup, tmpErr := MakeTmpFile(tenMBInBytes)
+	require.NoError(t, tmpErr)
+	defer func() {
+		cleanupErr := cleanup()
+		require.NoError(t, cleanupErr)
+	}()
+
+	reader := NewWrappedReader(tmpFile)
```

Comment:
I don't understand: why do I need to create a 10-MiB file if I'm not going to read all 10 MiB of it?  Can't I just implement a reader that makes up blocks, say blocks of zeroes?